Add segmentation data publisher to camera plugin

dscroft opened the pull request:
Segmentation topic will appear for Camera nodes that have segmentation available. Data will only be published if subscribers are connected.

**Description**
Camera plugin does not have segmentation data available.

**Related Issues**
N/A

**Affected Packages**
List of affected packages:
  - webots_ros2
  - webots_ros2_driver

**Tasks**
NA

**Additional context**
NA

Commit:
Add segmentation data publisher to camera plugin

Segmentation topic will appear for Camera nodes that have segmentation available.
Data will only be published if subscribers are connected.

Commit:
Segmentation requires recognition to be enabled

Fixed issue where segmentation did not work unless recognition was also subscribed

lukicdarkoo committed:
Merge branch 'master' into master

lukicdarkoo commented:
@dscroft Thank you. I synced with the main branch and allowed CI.

It looks good. Could you please create a simple test and add a changelog.

Commit:
Added segmentation test for webots_driver.

World has been slightly modified to enable segmentation on the camera.
Test checks for width and height of recieved image, black pixels in the image (no object), red pixels in the image (object segmentation highlight).

Commit:
Merge branch 'master' of github.com:dscroft/webots_ros2

dscroft commented:
Test added

omichel committed:
Merge branch 'master' into master

Commit:
Fixed issue in testSegmentation.

Segmentation data is not available in first frame so set comparison was failing.

dscroft commented:
Found the issue, pushed fix.

lukicdarkoo committed:
Merge branch 'master' into master

dscroft committed:
Merge branch 'cyberbotics:master' into master

dscroft committed:
Merge branch 'cyberbotics:master' into master
